src/vs/workbench/contrib/chat/common/chatEditingService.ts

```diff
@@ -110,19 +118,134 @@ export const enum ChatEditingSessionChangeType {
+	readonly viewType: string;
+	readonly originalToCurrentEdits: Record<number, IOffsetEdit>;
+	readonly diffInfo: ICellDiffInfo[];
+}
+export type ISnapshotEntryDTO = ITextSnapshotEntryDTO | INotebookSnapshotEntryDTO;
+
+interface IModifiedEntryTelemetryInfoDTO {
+	readonly requestId: string;
+	readonly agentId?: string;
+	readonly command?: string;
+}
+
+
+export interface IModifiedEntryTelemetryInfo {
+	readonly agentId: string | undefined;
+	readonly command: string | undefined;
+	readonly sessionId: string;
+	readonly requestId: string;
+	readonly result: IChatAgentResult | undefined;
+}
+
+interface IModifiedAnyFileEntry extends IDisposable {
+	readonly entryId: string;
 	readonly originalURI: URI;
-	readonly originalModel: ITextModel;
 	readonly modifiedURI: URI;
+	readonly onDidDelete: Event<void>;
 	readonly state: IObservable<WorkingSetEntryState>;
 	readonly isCurrentlyBeingModified: IObservable<boolean>;
 	readonly rewriteRatio: IObservable<number>;
-	readonly diffInfo: IObservable<IDocumentDiff>;
 	readonly lastModifyingRequestId: string;
+	readonly telemetryInfo: IModifiedEntryTelemetryInfo;
 	accept(transaction: ITransaction | undefined): Promise<void>;
 	reject(transaction: ITransaction | undefined): Promise<void>;
+	acceptStreamingEditsStart(tx: ITransaction): void;
+	acceptStreamingEditsEnd(tx: ITransaction): void;
+	updateTelemetryInfo(telemetryInfo: IModifiedEntryTelemetryInfo): void;
+	resetToInitialValue(): Promise<void>;
 }
 
+export type ICellDiffInfo = {
+	originalCellIndex: number;
+	modifiedCellIndex: number;
+	type: 'unchanged';
+} | {
+	originalCellIndex: number;
+	modifiedCellIndex: number;
+	type: 'modified';
+	diff: IDocumentDiff;
+} |
+{
+	originalCellIndex: number;
+	type: 'delete';
+} |
+{
+	modifiedCellIndex: number;
+	type: 'insert';
+};
+
+
+export interface IModifiedTextFileEntry extends IModifiedAnyFileEntry {
+	readonly kind: 'text';
+	readonly originalModel: ITextModel;
+	readonly modifiedModel: ITextModel;
+	readonly initialContent: string;
+	readonly diffInfo: IObservable<IDocumentDiff>;
+	acceptAgentEdits(textEdits: TextEdit[], isLastEdits: boolean): void;
+	createSnapshot(requestId: string | undefined): Promise<ITextSnapshotEntry>;
+	restoreFromSnapshot(snapshot: ITextSnapshotEntry): Promise<void>;
+}
+
+export interface IModifiedNotebookFileEntry extends IModifiedAnyFileEntry {
+	readonly kind: 'notebook';
+	readonly viewType: string;
+	readonly originalModel: INotebookTextModel;
+	readonly modifiedModel: INotebookTextModel;
+	readonly diffInfo: IObservable<ICellDiffInfo[]>;
+	acceptAgentCellEdits(cellUri: URI, textEdits: TextEdit[], isLastEdits: boolean): void;
+	acceptAgentNotebookEdits(edits: ICellEditOperation[], isLastEdits: boolean): Promise<void>;
+	createSnapshot(requestId: string | undefined): Promise<INotebookSnapshotEntry>;
+	restoreFromSnapshot(snapshot: INotebookSnapshotEntry): Promise<void>;
+}
+
+export type IModifiedFileEntry = IModifiedTextFileEntry | IModifiedNotebookFileEntry;
```

Comment:
This is the crux of the PR, the split into two distinct interfaces, one for text and one for notebooks

Reply:
Separate PR to create class implementing the notebook interface